Sources/GDBRemote/DebugSessionImpl.cpp

```diff
@@ -942,28 +942,54 @@ DebugSessionImpl::onResume(Session &session,
     }
   }
 
-  //
   // If kErrorAlreadyExist is set, then a signal is already pending.
-  //
   if (error != kErrorAlreadyExist) {
-    //
-    // Wait for the next signal
-    //
-    error = _process->wait();
-    if (error != kSuccess)
-      goto ret;
+    for (;;) {
+      error = _process->wait();
+      if (error != kSuccess) {
+        goto ret;
+      }
+
+      auto thread = _process->currentThread();
+      if (thread == nullptr) {
+        break;
+      }
+
+      if (thread->stopInfo().event != StopInfo::kEventStop) {
+        break;
+      }
+
+      switch (thread->stopInfo().reason) {
```

Comment:
Look at the commit message.

On Monday, April 18, 2016, walter erquinigo <notifications@github.com>
wrote:

> In Sources/GDBRemote/DebugSessionImpl.cpp
> <https://github.com/facebook/ds2/pull/283#discussion_r60162672>:
>
> > +    for (;;) {
> > +      error = _process->wait();
> > +      if (error != kSuccess) {
> > +        goto ret;
> > +      }
> > +
> > +      auto thread = _process->currentThread();
> > +      if (thread == nullptr) {
> > +        break;
> > +      }
> > +
> > +      if (thread->stopInfo().event != StopInfo::kEventStop) {
> > +        break;
> > +      }
> > +
> > +      switch (thread->stopInfo().reason) {
>
> this looks weird to me, why don't you just use an if inside the #if?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/283/files/63dabce5f93f14243f6ed214d6283b97e23f291a#r60162672>
>


-- 
Stephane Sezer